File: 2020/day1/python/part1.py
#!/usr/bin/env python3

if __name__ == "__main__":

    with open("../input.txt", "r") as fh:
        lines = fh.readlines()
    numbers = [int(line, 10) for line in lines]

    for number in numbers:
        sought = 2020 - number
        if sought in numbers:
            print(sought * number)
            break

File: 2020/day1/perl/part1.pl
#!/usr/bin/perl

use strict;
use warnings;

open(my $fh, "<", "../input.txt") or die $!;

my @lines = <$fh>;
chomp @lines;

for (my $i=0; $i < $#lines; $i++) {
    my $n = $lines[$i];
    for (my $j=$i+1; $j < $#lines; $j++) {
        my $m = $lines[$j];
        if ($n + $m == 2020) {
            print $n*$m . "\n";
        }
    }
}
